src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemDatasetValidator.Cancel.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Globalization;
+using EnsureThat;
+using FellowOakDicom;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Store;
+using Microsoft.Health.Dicom.Core.Features.Workitem.Model;
+
+namespace Microsoft.Health.Dicom.Core.Features.Workitem
+{
+    public sealed class CancelWorkitemDatasetValidator : WorkitemDatasetValidator
+    {
+        protected override void OnValidate(DicomDataset dicomDataset, string workitemInstanceUid)
+        {
+            // all the attributes in the dataset are optional. Hence no validation is being done here.
+        }
+
+        public static void ValidateProcedureStepStateInStore(string workitemUid, WorkitemMetadataStoreEntry workitemMetadata, WorkitemStateTransitionResult stateTransitionResult)
+        {
+            EnsureArg.IsNotNull(stateTransitionResult, nameof(stateTransitionResult));
+
+            if (workitemMetadata == null)
+            {
+                throw new WorkitemNotFoundException(workitemUid);
+            }
+
+            if (workitemMetadata.Status != WorkitemStoreStatus.ReadWrite)
+            {
+                throw new DatasetValidationException(
```

Comment:
For this case, since we are attempting to change state, not strictly update - why don't we return DicomCoreResource.InvalidProcedureStepStateTransition with the error code C313 https://dicom.nema.org/medical/dicom/current/output/html/part04.html#table_CC.2.2-2